external/vcm/vcm/data_transform.py

```diff
@@ -0,0 +1,83 @@
+from collections import defaultdict
+import dataclasses
+from toolz.functoolz import curry
+from typing import Callable, Union, Literal, Sequence
+import xarray as xr
+import vcm
+
+DATA_TRANSFORM_FUNCTIONS = {}
+
+
+@curry
+def register(
+    name: str, func: Callable[..., xr.Dataset], registry=DATA_TRANSFORM_FUNCTIONS
+):
+    if name in registry:
+        raise ValueError(f"Function {name} has already been added to registry.")
+    registry[name] = func
+    return func
+
+
+@register("qm_from_q1_q2")
+def qm_from_q1_q2(ds, temperature_dependent_latent_heat=False):
+    if temperature_dependent_latent_heat:
+        Qm = vcm.moist_static_energy_tendency(
+            ds["Q1"], ds["Q2"], temperature=ds["air_temperature"]
+        )
+    else:
+        Qm = vcm.moist_static_energy_tendency(ds["Q1"], ds["Q2"])
+    ds["Qm"] = Qm
+    return ds
+
+
+@register("q1_from_qm_q2")
+def q1_from_qm_q2(ds, temperature_dependent_latent_heat=False):
+    if temperature_dependent_latent_heat:
+        Q1 = vcm.temperature_tendency(
+            ds["Qm"], ds["Q2"], temperature=ds["air_temperature"]
+        )
+    else:
+        Q1 = vcm.temperature_tendency(ds["Qm"], ds["Q2"])
+    ds["Q1"] = Q1
+    return ds
+
+
+@register("q1_from_dQ1_pQ1")
+def q1_from_dQ1_pQ1(ds):
+    ds["Q1"] = ds["dQ1"] + ds["pQ1"]
+    return ds
+
+
+@register("q2_from_dQ2_pQ2")
+def q2_from_dQ2_pQ2(ds):
+    ds["Q2"] = ds["dQ2"] + ds["pQ2"]
+    return ds
+
+
+@dataclasses.dataclass
+class DataTransformConfig:
+    name: Literal[
+        "q1_from_qm_q2", "qm_from_q1_q2", "q1_from_dQ1_pQ1", "q2_from_dQ2_pQ2"
```

Comment:
Some duplication, since need to add transformation name here in addition to registering it above. But I think worth it for the validation we'll get from `dacite`.